integration/spark/src/main/common/java/io/openlineage/spark/agent/util/PathUtils.java

```diff
@@ -63,19 +64,46 @@ public static DatasetIdentifier fromCatalogTable(CatalogTable catalogTable, Stri
     }
   }
 
+  @SneakyThrows
   public static DatasetIdentifier fromCatalogTable(CatalogTable catalogTable) {
     String authority =
         ScalaConversionUtils.asJavaOptional(catalogTable.storage().locationUri())
             .map(URI::toString)
-            .orElse(
-                SparkSession.active()
-                    .sessionState()
-                    .catalog()
-                    .defaultTablePath(catalogTable.identifier())
-                    .toString());
+            .orElseGet(
+                () ->
+                    SparkSession.active()
+                        .sessionState()
+                        .catalog()
+                        .defaultTablePath(catalogTable.identifier())
+                        .toString());
     return PathUtils.fromCatalogTable(catalogTable, authority);
   }
 
+  @SneakyThrows
+  public static DatasetIdentifier fromHiveTable(
```

Comment:
Add a javadoc comment here explaining the reasoning behind this impl. I.e., the thinking behind symlinking datasets based on facets